test/avocado/machineslib.py

```diff
@@ -0,0 +1,166 @@
+  <features>
+    <acpi/>
+  </features>
+  <cpu mode='host-model'>
+    <model fallback='forbid'/>
+  </cpu>
+  <devices>
+    <disk type='file' device='disk'>
+      <driver name='qemu' type='qcow2'/>
+      <source file='{image}'/>
+      <target dev='hda' bus='ide'/>
+    </disk>
+    <controller type='scsi' model='virtio-scsi' index='0' id='hot'/>
+    <interface type='network'>
+      <source network='default' bridge='virbr0'/>
+      <target dev='vnet0'/>
+    </interface>
+    {console}
+    {graphics}
+  </devices>
+</domain>
+"""
+
+
+class MachinesLib(SeleniumTest):
+    """
+    :avocado: disable
+    """
+
+    def create_vm(self, name, graphics='spice', ptyconsole=False, state='running', wait=False):
+        self.virshvm = name
+
+        img = "/var/lib/libvirt/images/cirros.qcow2"
+        pool_name = "images"
+        pool_path = os.path.dirname(img)
+        self.machine.execute("test -f {}".format(img))
+        self.machine.execute(
+            "sudo virsh pool-list --all | grep {pool} || sudo virsh pool-create-as {pool} --type dir --target {path} && sudo virsh pool-refresh {pool}".format(pool=pool_name, path=pool_path))
+
+        args = {
+            "name": name,
+            "poolName": pool_name,
+            "poolPath": pool_path,
+            "image": img,
+            "logfile": None,
+            "console": "",
+            "graphics": ""
+        }
+
+        if ptyconsole:
+            args["console"] = PTYCONSOLE_XML
+        else:
+            self.machine.execute("sudo chmod 777 /var/log/libvirt")
+            args["logfile"] = "/var/log/libvirt/console-{0}.log".format(name)
+            args["console"] = CONSOLE_XML.format(log=args["logfile"])
+
+        if graphics == 'spice':
+            cxml = SPICE_XML
+        elif graphics == 'vnc':
+            cxml = VNC_XML
+        else:
+            cxml = ""
+        args["graphics"] = cxml
+
+        xml = DOMAIN_XML.format(**args)
+        self.machine.execute('sudo echo \"{}\" > /tmp/xml && sudo virsh define /tmp/xml'.format(xml))
+        if state == 'running':
+            self.machine.execute('sudo virsh start {}'.format(name))
+            if wait:
+                self.wait_vm_complete_start(args)
+        elif state == 'shut off':
+            pass
+
+        self.wait_in_text(name, '#vm-{}-row'.format(name))
+        self.wait_in_text(state, '#vm-{}-state'.format(name))
+        self.click(self.wait_css("tbody tr[data-row-id='vm-{}'] th".format(name), cond=clickable))
+
+        return args
+
+    def wait_vm_complete_start(self, vmargs):
+        log_file = vmargs.get('logfile')
+        if log_file != None:
+            search_str = "login as 'cirros' user."
+            cmd = 'until sudo cat {} | grep "{}"; do sleep 1; done'.format(log_file, search_str)
+            # self.machine.execute(script=WAIT_VM.format(cmd))
+            self.machine.execute(cmd)
+        else:
+            time.sleep(10)
+
+    def destroy_vm(self, name):
+        try:
```

Comment:
could be possible to do this exception less generic, eg. just catch called process error, or try except just some part what fails randomly.

Reply:
Makes sense, but I think it's better to raise the exception when failed to delete the vm as it will affect the next test. So I'd delete the exception capture part.